pep-0xxx.rst

```diff
@@ -0,0 +1,727 @@
+  the standard library understands, forcing projects that want to use them to
+  maintain substantial compatibility layers.
+
+* For Windows distributions of Python, they need to be shipped with a copy of
+  OpenSSL. This puts the CPython development team in the position of being
+  OpenSSL redistributors, potentially needing to ship security updates to the
+  Windows Python distributions when OpenSSL vulnerabilities are released.
+
+* For macOS distributions of Python, they need either to be shipped with a copy
+  of OpenSSL or linked against the system OpenSSL library. Apple has formally
+  deprecated linking against the system OpenSSL library, and even if they had
+  not, that library version has been unsupported by upstream for nearly one
+  year as of the time of writing. The CPython development team has started
+  shipping newer OpenSSLs with the Python available from python.org, but this
+  has the same problem as with Windows.
+
+* Many systems, including but not limited to Windows and macOS, do not make
+  their system certificate stores available to OpenSSL. This forces users to
+  either obtain their trust roots from elsewhere (e.g. `certifi`_) or to
+  attempt to export their system trust stores in some form.
+
+  Relying on `certifi`_ is less than ideal, as most system administrators do
+  not expect to receive security-critical software updates from PyPI.
+  Additionally, it is not easy to extend the `certifi`_ trust bundle to include
+  custom roots, or to centrally manage trust using the `certifi`_ model.
+
+  Even in situations where the system certificate stores are made available to
+  OpenSSL in some form, the experience is still sub-standard, as OpenSSL will
+  perform different validation checks than the platform-native TLS
+  implementation. This can lead to users experiencing different behaviour on
+  their browsers or other platform-native tools than they experience in Python,
+  with little or no recourse to resolve the problem.
+
+* Users may wish to integrate with TLS libraries other than OpenSSL for many
+  other reasons, such as OpenSSL missing features (e.g. TLS 1.3 support), or
+  because OpenSSL is simply too large and unweildy for the platform (e.g. for
+  embedded Python). Those users are left with the requirement to use
+  third-party networking libraries that can interact with their preferred TLS
+  library or to shim their preferred library into the OpenSSL-specific ``ssl``
+  module API.
+
+Additionally, the ``ssl`` module as implemented today limits the ability of
+CPython itself to add support for alternative TLS backends, or remove OpenSSL
+support entirely, should either of these become necessary or useful. The
+``ssl`` module exposes too many OpenSSL-specific function calls and features to
+easily map to an alternative TLS backend.
+
+
+Proposal
+========
+
+This PEP proposes to introduce a few new Abstract Base Classes in Python 3.7 to
```

Comment:
Also you probably want to indicate that the `ssl` module will implement these intefaces :-)